src/Contrib/Jaeger/SpanConverter.php

```diff
@@ -174,28 +173,88 @@ private static function convertOtelSpanDataToJaegerTags(SpanDataInterface $span)
         }
 
         foreach ($span->getAttributes() as $k => $v) {
-            $tags[$k] = self::sanitiseTagValue($v);
+            $tags[$k] = $v;
         }
         $tags = self::buildTags($tags);
 
         return $tags;
     }
 
-    private static function sanitiseTagValue($value): string
+    private static function buildTags(array $tagPairs): array
+    {
+        $tags = [];
+        foreach ($tagPairs as $key => $value) {
+            $tags[] = self::buildTag($key, $value);
+        }
+
+        return $tags;
+    }
+
+    private static function buildTag(string $key, $value): Tag
+    {
+        $valueToCreateTagFrom = self::convertValueToTypeJaegerTagsSupport($value);
+
+        return self::createJaegerTagInstance($key, $valueToCreateTagFrom);
+    }
+
+    private static function convertValueToTypeJaegerTagsSupport($value)
+    {
+        if (is_array($value)) {
+            return self::serializeArrayToString($value);
+        }
+
+        return $value;
+    }
+
+    private static function createJaegerTagInstance(string $key, $value)
     {
-        // Casting false to string makes an empty string
         if (is_bool($value)) {
-            return $value ? 'true' : 'false';
+            return new Tag([
+                'key' => $key,
+                'vType' => TagType::BOOL,
+                'vBool' => $value,
+            ]);
+        } elseif (is_integer($value)) {
+            return new Tag([
+                'key' => $key,
+                'vType' => TagType::LONG,
+                'vLong' => $value,
+            ]);
+        } elseif (is_numeric($value)) {
+            return new Tag([
+                'key' => $key,
+                'vType' => TagType::DOUBLE,
+                'vDouble' => $value,
+            ]);
```

Comment:
The Else-Police says: In this case else is really not needed, since you are returning values.
```php
        if (is_bool($value)) {
            return new Tag([
                'key' => $key,
                'vType' => TagType::BOOL,
                'vBool' => $value,
            ]);
        } 
        if (is_integer($value)) {
            return new Tag([
                'key' => $key,
                'vType' => TagType::LONG,
                'vLong' => $value,
            ]);
        } 
        if (is_numeric($value)) {
            return new Tag([
                'key' => $key,
                'vType' => TagType::DOUBLE,
                'vDouble' => $value,
            ]);
        }
```

Reply:
Very much agreed, updated!